docgen/src/guides/calendar-widget.md

```diff
@@ -0,0 +1,198 @@
+
+We need to add the dependencies at the end of the `body`.
+
+```html
+<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/moment.js/2.20.1/moment.min.js"></script>
+<script src="https://unpkg.com/BaremetricsCalendar@1.0.11/public/js/Calendar.js"></script>
+```
+
+In order for the Baremetrics Calendar to work, we need to define a `#calendar` element that is going to be attached the calendar itself:
+
+```html
+<div id="calendar" class="daterange"></div>
+```
+
+We'll attach the calendar to the element in the next section.
+
+### Creating our custom widget
+
+InstantSearch doesn't provide any calendar widgets. Yet, there are two widgets handling numeric ranges: [`rangeInput`](https://community.algolia.com/instantsearch.js/v2/widgets/rangeInput.html) and [`rangeSlider`](https://community.algolia.com/instantsearch.js/v2/widgets/rangeSlider.html). If we were to use these widgets, we wouldn't be able to customize their rendering. Both of these widgets are based on the [`connectRange`](https://community.algolia.com/instantsearch.js/v2/connectors/connectRange.html) connector which handles their logic. Because the dates are stored as numeric values in our dataset, we are going to use this connector to create our custom calendar widget.
+
+If you are not yet familiar with [connectors](https://community.algolia.com/instantsearch.js/v2/connectors.html), you can think of them as the logic-only part of your widget. It doesn't render anything -- that's what makes them reusable.
+
+A connector takes a rendering function which is called every time the search is refined. The calendar library we're using only needs to be initialized once. Luckily, the second parameter of the `connectRange` callback function is `isFirstRendering`. We, therefore, can instantiate the calendar only at the first rendering.
+
+```javascript
+const makeRangeWidget = instantsearch.connectors.connectRange(
+  (options, isFirstRendering) => {
+    if (!isFirstRendering) return;
+
+    /* bind the calendar to the connector here */
+  }
+);
+```
+
+`connectRange` is passed as its first argument's callback a [`RangeRenderingOptions`](https://community.algolia.com/instantsearch.js/v2/connectors/connectRange.html#struct-RangeRenderingOptions) object which contains the `refine(Array<number,number>)` function. The two numbers that need to be passed are the start and the end dates as numeric timestamps.
+
+In our connector instantiation, we need to attach the calendar and specify its callback function to refine the search. This part will be specific to the calendar library that you use in your app. In the Baremetrics Calendar, this function is passed to the `callback` argument.
+
+```javascript
+const makeRangeWidget = instantsearch.connectors.connectRange(
```

Comment:
proposal: call the widget `datePicker`, since that's more consistent with the names in IS.js

Reply:
Yeah, it's also more explicit!